### PR TITLE
Trending slopes

### PR DESCRIPTION
**Pull request by skhrg**
As discussed on slack this uses slope instead of max-min to find trending detectors.
The value for `max_trend` is from a quick look at some data from last year. As I look at the data from the most recent LATR cooldown I will refine the default.

### CODE REVIEW

File: sotodlib/flags.py [hunk 1]
```diff
@@ -82,115 +85,136 @@ def get_glitch_flags(
...
+            s = s[:, ::n]
+        _slopes = ((t * s).mean(axis=1) - t.mean() * s.mean(axis=1)) / (
+            (t**2).mean() - (t.mean()) ** 2
+        )
+        cut = np.hstack(
+            (cut, np.tile((np.abs(_slopes) > max_trend)[..., np.newaxis], samps))
+        )
+        if full_output:
+            slopes = np.hstack((slopes, _slopes[..., np.newaxis]))
+            samp_edges.append(samp_edges[-1] + len(t))
```
Comment (kmharrington) [inline sotodlib/flags.py:197]: `len(t)` needs to be `samps` here in case the array has been cut down

File: sotodlib/flags.py [hunk 2]
```diff
@@ -82,115 +85,136 @@ def get_glitch_flags(
...
+
+    slopes = np.zeros((len(signal), 0))
+    cut = np.zeros((len(signal), 0), dtype=bool)
+    samp_edges = [0]
+    for t, s in zip(
+        np.array_split(timestamps, n_pieces), np.array_split(signal, n_pieces, 1)
+    ):
+        samps = len(t)
+        # Cheap downsampling
+        if len(t) < max_samples:
```
Comment (kmharrington) [inline sotodlib/flags.py:185]: this should be if samps > max_samples. We don't need to down sample if the array is shorter

Comment (skhrg) [reply sotodlib/flags.py:185]: Good catch

**Review by kmharrington — CHANGES_REQUESTED** [review]
The two inline comments look like it wasn't fully tested for cases where the arrays are longer or shorter than max samples. Those definitely need to be fixed. 

While we're at it, could I ask you to write test cases for this function? We have zero coverage here and I had to stare at this enough that it might be a good place to start it.

**Comment by skhrg** [comment]
Is there a specific place I should put the test? I don't see a `test_flags.py`

**Comment by kmharrington** [comment]
yeah go ahead and make it!

**Comment by skhrg** [comment]
@kmharrington  let me know if you want the test to cover a wider scope of cases.

**Review by kmharrington — APPROVED** [review]
Looks pretty good to me. One more test case would be running with n_pieces > 1. Make sure that all gets the right things as well.

**Pull request merged**